src/app/frontend/common/components/resourcelist/statefulset/component.ts

```diff
@@ -74,24 +73,30 @@ export class StatefulSetListComponent extends ResourceListWithStatuses<
     return statefulSetList.statefulSets;
   }
 
-  hasUnhealthyPods(resource: StatefulSet): boolean {
-    return resource.podInfo.running !== resource.podInfo.desired;
-  }
-
   isInErrorState(resource: StatefulSet): boolean {
     return resource.podInfo.warnings.length > 0;
   }
 
   isInPendingState(resource: StatefulSet): boolean {
-    return resource.podInfo.warnings.length === 0 && resource.podInfo.pending > 0;
+    return (resource.podInfo.warnings.length === 0 && resource.podInfo.pending > 0) || (resource.podInfo.running !== resource.podInfo.desired);
```

Comment:
Pending and desired/running part should be taken into common parenthesis. It should be

_if there are no warnings and (there are pending pods or running !== desired)_

Adjust success state check accordingly.